src/frontend/src/eth/services/send.services.ts

```diff
@@ -508,14 +511,45 @@ const approve = async ({
 			erc20HelperContractAddress
 		})
 	) {
-		return { transactionApproved: false };
+		return { transactionApproved: false, nonce };
 	}
 
+	const preApprovedAmount = await erc20ContractAllowance({
+		token,
+		owner: from,
+		spender: erc20HelperContractAddress,
+		networkId
+	});
+
+	// If there is already an approved allowance that is enough for the required amount, we don't need to approve again.
+	if (preApprovedAmount.gte(amount)) {
+		progress(ProgressStepsSend.APPROVE);
+		return { transactionApproved: false, nonce };
```

Comment:
I'm confused about `transactionApproved: false` because I interpret it as if it were an error. I'm also confused about whether we should distinguish between this state and, although not yet used, a potential `success: false` in `prepareAndSignApproval`.

Do you know what I mean?

Reply:
Yes, not that clear indeed: I wanted to keep the same logic in the sense that `transactionApproved` indicates that we actually created an "approval" transaction. For example, if the function is run for something that it is not ck-conversion, the code returns `transactionApproved: false`, meaning that there was no need to approve anything, nor create a transaction to approve anything. But that does not indicates an error, it is just used to increase the `nonce` or not.

I would suggest to rename it as `approvalTransactionNeeded`, or something similar. Or remove it completely and just use a nullish check on the `hash` return: if hash is undefined (no new approval transaction), then do not increment the nonce

Reply:
Check this one please, does it make sense?

https://github.com/dfinity/oisy-wallet/pull/3004